### PR TITLE
handle more transitions for self-closing tags and bump up version

### PR DESCRIPTION
**Pull request by maditya**
@neraliu @adon-at-work please review. 
This addresses https://github.com/yahoo/html-purify/issues/3

### CODE REVIEW

File: tests/test-vectors.js [hunk 1]
```diff
@@ -765,8 +765,27 @@ var generalVectors = [
 	id: 37,
 	input: "<div style=\"color:))))) green\">parenthesis test 5</div>",
 	output: "<div>parenthesis test 5</div>"
+},
+{
+	id: 38,
+	input: "<img />",
```
Comment (neraliu) [inline tests/test-vectors.js:771]: u need one more test case - `` and `` are different according to the spec.

File: tests/test-vectors.js [hunk 2]
```diff
@@ -765,8 +765,27 @@ var generalVectors = [
...
+	output: "<img />"
+},
+{
+	id: 39,
+	input: "<img id=\"foo\" />",
```
Comment (neraliu) [inline tests/test-vectors.js:776]: same here - ``, ``, ``

File: src/html-purify.js [hunk 3]
```diff
@@ -92,7 +95,7 @@ See the accompanying LICENSE file for terms.
             break;
 
         case derivedState.TransitionName.ATTR_TO_AFTER_ATTR:
-            this.attrVals[attributeName] = '';
+            this.attrVals[attributeName] = null;
```
Comment (maditya) [inline src/html-purify.js]: @adon-at-work @neraliu should we also initialize attributeValue to null as opposed to '' at https://github.com/yahoo/context-parser/blob/master/src/context-parser.js#L43 ?

File: src/html-purify.js [hunk 4]
```diff
@@ -52,23 +52,26 @@ See the accompanying LICENSE file for terms.
...
 
         case derivedState.TransitionName.FROM_TAG_ATTR_TO_DATA:
-            if (prevState === 35 ||
-                prevState === 36 || 
+            if (prevState === 36 || 
                 prevState === 40) {
                 this.attrVals[attributeName] = attributeValue;
             }
+            // boolean attributes may not have a value
+            if (prevState === 35) {
```
Comment (maditya) [inline src/html-purify.js]: FYI - I am also adding a list of boolean attributes. We will set new value of non-boolean attribute with no value to ""(in a new PR)

File: src/html-purify.js (hunk 3 above)
Comment (adon-at-work) [inline src/html-purify.js]: @yukinying has an answer for you

File: src/html-purify.js (hunk 4 above)
Comment (adon-at-work) [inline src/html-purify.js]: i see now why you need to differentiate `null` from `''`

File: src/html-purify.js (hunk 3 above)
Comment (yukinying) [inline src/html-purify.js]: ~~I would prefer `''` as it allows appending via `+` operator.~~

Looking at another comment below (attribute name without attribute value), I think we may need a fix upstream to initialize attribute values as null in context parser.

**Comment by adon-at-work** [comment]
any updates, @maditya / @neraliu ?

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by neraliu** [comment]
one comment, do we need to have the CP PR for setting attribute null before merging this?

**Comment by adon-at-work** [comment]
released the latest changes at https://github.com/yahoo/context-parser/pull/36
please rebase, and update package.json to use the latest one.

**Pull request merged**